test/parallel/test-require-resolve.js

```diff
@@ -80,3 +80,16 @@ require(fixtures.path('resolve-paths', 'default', 'verify-paths.js'));
     assert.strictEqual(resolvedPaths.includes('/node_modules'), false);
   });
 }
+
+{
+  assert.strictEqual(require.resolve('node:test'), 'node:test');
+  assert.strictEqual(require.resolve('node:fs'), 'node:fs');
+
+  assert.throws(
+    () => require.resolve('node:unknown'),
+    {
+      code: 'MODULE_NOT_FOUND',
+      message: /^Cannot find module 'node:unknown'/,
+    },
```

Comment:
We should not validate the `message` property, only the code is guaranteed by the docs (and it's ok to change error messages as a non-breaking change, but not error codes).
```suggestion
    { code: 'MODULE_NOT_FOUND' },
```